### PR TITLE
TRUNK-5499-Upgrading the Hibernate Library from 4.3.9.Final to 4.3.10.Final

### PR DESCRIPTION
**Pull request by Irenyak1**
## Description of what I changed

I upgraded the Hibernate  version from 4.3.9.Final to 4.3.10.Final
thus  upgrading the following 3 Sub Libraries

org.hibernate:hibernate-c3p0 … 4.3.9.Final -> 4.3.10.Final

org.hibernate:hibernate-core … 4.3.9.Final -> 4.3.10.Final

org.hibernate:hibernate-ehcache … 4.3.9.Final ->  4.3.10.Final

also I upgraded

org.hibernate:hibernate-validator … 4.2.0.Final -> 6.0.16.Final
org.hibernate:hibernate-search-orm … 5.1.2.Final -> 5.3.0.Final

## Issue I worked on
I have worked on this issue 
https://issues.openmrs.org/browse/TRUNK-5499

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1047,7 +1047,7 @@
 		<openmrs.version>${project.version}</openmrs.version>
 
 		<springVersion>4.1.4.RELEASE</springVersion>
-		<hibernateVersion>4.3.9.Final</hibernateVersion>
+		<hibernateVersion>4.3.10.Final</hibernateVersion>
```
Comment (HerbertYiga) [inline pom.xml]: are you updating to the latest hibernate version,and if so is it really this??

Comment (Irenyak1) [reply pom.xml]: @HerbertYiga you may look through [this](https://talk.openmrs.org/t/there-seems-to-be-a-backward-incompatibility-in-hibernate-versions/23401/18)

File: api/pom.xml [hunk 2]
```diff
@@ -209,6 +209,11 @@
 			<groupId>org.hibernate</groupId>
 			<artifactId>hibernate-validator</artifactId>
 	   </dependency>
+	   <dependency>
+			<groupId>mysql</groupId>
```
Comment (dkayiwa) [inline api/pom.xml]: Is this part of your changes?

Comment (Irenyak1) [reply api/pom.xml]: @dkayiwa no it is not

Comment (dkayiwa) [reply api/pom.xml]: Then why is it here?

Comment (Irenyak1) [reply api/pom.xml]: I don't know how it got mixed up with my PR but I see it's you @dkayiwa  and @reagan-meant  who did this commit.

Comment (reagan-meant) [reply api/pom.xml]: Did you create the branch off of master? Hope not on my trunk

Comment (Irenyak1) [reply api/pom.xml]: I created off master @reagan-meant

Comment (gitcliff) [reply api/pom.xml]: @Irenyak1 you need to squash the commits into one

Comment (Irenyak1) [reply api/pom.xml]: @gitcliff yeah sure.

**Comment by mozzy11** [comment]
ping @dkayiwa , thislooks pretty fine for now.

**Comment by Irenyak1** [comment]
@dkayiwa  were you able to look through this PR, is it ready for merging or are there any issues I need to address first? Let me know.

**Comment by dkayiwa** [comment]
@Irenyak1 your pull request description does not match with the actual changes.

**Comment by Irenyak1** [comment]
@dkayiwa  does the PR description look better now? I have edited it.

**Comment by Irenyak1** [comment]
hello @dkayiwa is there still something stopping this to be merged?

**Comment by dkayiwa** [comment]
What is in the pull request does not match what is on the ticket.

**Comment by Irenyak1** [comment]
Hello @dkayiwa  I seek for clarity on your comment. Is it the PR title which does not match the ticket title or it is the PR description which does not match the ticket description? 

Thank you.

**Comment by dkayiwa** [comment]
Did you take a second look at what is on the ticket?

**Comment by Irenyak1** [comment]
@dkayiwa  yes I did. The title of the PR is different from the Ticket title because the other time I inquired from you, you told me the PR title should reflect what I have done in the ticket.

**Comment by dkayiwa** [comment]
Can you ask for help?

**Comment by Irenyak1** [comment]
@dkayiwa  the description is not similar to the ticket's because I did not upgrade to the exact versions per the ticket but to working versions.

**Comment by dkayiwa** [comment]
Can some one tell that from the ticket?

**Comment by Irenyak1** [comment]
@dkayiwa  what was I supposed to have done to make the PR understandable?

**Comment by Irenyak1** [comment]
@dkayiwa  yeah sure you can help me out.

**Comment by dkayiwa** [comment]
If one looks at the ticket, how does he or she tell why your pull request does not do what the tickets says?

**Comment by Irenyak1** [comment]
@dkayiwa  so that means I have to put a detailed description of the change in the PR showing what the ticket was meant to do and now what the PR does? Is that the way I should have done?

**Comment by Irenyak1** [comment]
@dkayiwa  does that mean I only need to change the description only and leave the title as it is?

**Comment by dkayiwa** [comment]
Put a comment explaining why you did what you did.

**Comment by Irenyak1** [comment]
@dkayiwa should I put that comment  within the PR description, if not, where do I put it?

**Comment by dkayiwa** [comment]
On the ticket.

**Comment by Irenyak1** [comment]
@dkayiwa  ok thanks let me comment on the ticket. Is there anything I need to change or comment on the PR?

**Comment by Irenyak1** [comment]
@dkayiwa  I have commented on the ticket please could you review my comment and let me know if it is sufficient or not.

**Comment by dkayiwa** [comment]
Can you avoid repeating what is already in the ticket description?

**Comment by Irenyak1** [comment]
@dkayiwa  done you may check again. :)

**Comment by dkayiwa** [comment]
Did you take time to review your own comment before asking me to?

**Comment by Irenyak1** [comment]
@dkayiwa  yes I did review it twice.

**Comment by dkayiwa** [comment]
Review it again tomorrow when you are not sleepy!

**Comment by Irenyak1** [comment]
@dkayiwa  ok

**Comment by Irenyak1** [comment]
Hello @dkayiwa  you may review my PR, I edited the comment on the ticket. I hope it is good enough.

**Comment by dkayiwa** [comment]
Your ticket comment does not reflect the changes in your pull request. If you still can't figure it out, ask another person to help you.

**Comment by Irenyak1** [comment]
ok

**Comment by Irenyak1** [comment]
@mozzy11  please I am still waiting for your feedback on this. I need your help.

**Comment by Irenyak1** [comment]
Hello @dkayiwa  please review this PR. I have edited my comments on both the ticket and the PR. 

Thanks.

**Comment by mozzy11** [comment]
@Irenyak1  , Since you put a comment on the ticket explaining why you you made this change(not what the ticket says) , in the PR description , simply put what you did here , not the explanation why you didnt do what you were supposed to do. (Dont repeat what you already explained in the Ticket description)

also update you commit mesage to simplr refleft what you did

**Comment by Irenyak1** [comment]
Hello @dkayiwa and @mozzy11  I have worked on the comments and the commit message like you advised me earlier. Please review my PR.

**Comment by sherrif10** [comment]
@Irenyak1  the ticket is saying upgrading hibernate libraries from 4.3.9 to 5.4.3,  and you upgraded to
4.3.10

**Comment by Irenyak1** [comment]
@sherrif10  please read the comment on the ticket you will understand why it was upgraded to that. Have you read the comments on the PR and tickets first?

**Comment by dkayiwa** [comment]
@Irenyak1 shouldn't the ticket title change accordingly to reflect these changes?

**Comment by Irenyak1** [comment]
@dkayiwa  does this mean changing the ticket title from "Upgrading the Hibernate Library from 4.3.9.Final to 5.4.2.Final" to "Upgrading the Hibernate Library from 4.3.9.Final to 4.3.10.Final" 
in JIRA?

**Pull request merged**